torch/_ops.py

```diff
@@ -903,13 +970,45 @@ def __call__(self_, *args, **kwargs):  # noqa: B902
         # is still callable from JIT
         # We save the function ptr as the `op` attribute on
         # OpOverloadPacket to access it here.
+
+        # Directly calling OverloadPacket goes into C++, which will check
+        # the schema and cause an error for torchbind op when inputs consist of FakeScriptObject so we
+        # intercept it here and call TorchBindOpverload instead.
+        if self_._has_torchbind_op_overload and _must_dispatch_in_python(args, kwargs):
+            return _call_overload_packet_from_python(self_, args, kwargs)
         return self_._op(*args, **(kwargs or {}))
 
     # TODO: use this to make a __dir__
     def overloads(self):
         return [n if n else "default" for n in self._overload_names]
 
 
+# Note - this mirrors the logic of the cpp_function defined in jit/python/init.cpp
+# _jit_get_operations, which calls _get_operation_for_overload_or_packet.
+def _call_overload_packet_from_python(op: OpOverloadPacket, args, kwargs):
+    # Re-use the torch function handling logic in cpp
+    torch_function_called, ret = torch._C._maybe_call_torch_function_for_op_packet(
+        op, *args, **kwargs
+    )  # type: ignore[attr-defined]
+
+    if torch_function_called:
+        return ret
+
+    # In cpp, we do a schema matching for the arguments, and call ToIValue to
+    # to check whether the arguments are valid. But need to do similar things here
+    # and skip the ToIValue check because FakeScriptObject won't convert to IValue.
+    exceptions = {}
+    for overload_name in op.overloads():
+        try:
+            return getattr(op, overload_name)(*args, **kwargs)
+        except RuntimeError as e:
+            exceptions[overload_name] = e
+            continue
+    raise RuntimeError(
+        f"Fail to call all TorchBindOverloads of {op} with exceptions {exceptions}"
```

Comment:
I read through the C++ overloading code. This doesn't feel faithful enough:
- If the operator throws a RuntimeError, then we report it differently
- what happens if we have `foo(Tensor, TensorQueue)` and `foo(Tensor, AtomicCounter)`? Do ScriptObjects have overload resolution done on them?

It sounds like:
- we need some sort of `match_schema_ignoring_fake_object -> bool` C++ function. Or maybe we just need to construct a new schema, ignoring the script object, and then use that for schema matching.
- IF ScriptObjects have overload resolution done on them, then we need to model that